### PR TITLE
Close #DEVOPS-1719 add JSON formatter support

### PR DESCRIPTION
**Pull request by wshihadeh**
- Add support to JSON log format  request_headers_logger 
- Remove Tagged logging dependency 
- Fix duplicate tags issue

### CODE REVIEW

File: lib/request_headers_logger/json_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
```
Comment (MarcGrimme) [inline lib/request_headers_logger/json_formatter.rb:1]: We need tests here?

File: lib/request_headers_logger/json_formatter.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+module RequestHeadersLogger
+  class JsonFormatter
+    def call(severity, timestamp, progname, msg)
+      json = { level: severity, timestamp: timestamp.to_s, message: msg.strip }
+      json = json.merge(progname: progname.to_s) unless progname.nil?
+
+      json.merge(RequestHeadersLogger.tags || {}).to_json + "\n"
```
Comment (MarcGrimme) [inline lib/request_headers_logger/json_formatter.rb:9]: This is what I kinda don't really like without having a much better solution. But this is an implicit dep on ActiveSupport::TaggedLogger. and I don't think this is really needed like this for Json logging.

File: lib/request_headers_logger/text_formatter.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RequestHeadersLogger
+  class TextFormatter < ::Logger::Formatter
+    def call(severity, timestamp, progname, msg)
+      super(severity, timestamp, progname, "#{tags_text}#{msg}")
+    end
+
+    def tags_text
+      RequestHeadersLogger.tags.collect { |_key, tag| "[#{tag}] " }.join if RequestHeadersLogger.tags.any?
```
Comment (MarcGrimme) [inline lib/request_headers_logger/text_formatter.rb:10]: Same here but less problematic. Implicit dep the ActiveSupport::TaggedLogger.

File: lib/request_headers_logger/json_formatter.rb (hunk 1 above)
Comment (wshihadeh) [reply lib/request_headers_logger/json_formatter.rb:1]: testes were added

File: lib/request_headers_logger/json_formatter.rb (hunk 2 above)
Comment (wshihadeh) [reply lib/request_headers_logger/json_formatter.rb:9]: No this is not a dep on ActiveSupport::TaggedLogger.  

[`RequestHeadersLogger.tags`](https://github.com/fidor/request_headers_logger/blob/master/lib/request_headers_logger.rb#L14)  is just the store items after filtering the whitelisted ones.

File: lib/request_headers_logger/text_formatter.rb (hunk 3 above)
Comment (wshihadeh) [reply lib/request_headers_logger/text_formatter.rb:10]: [`RequestHeadersLogger.tags`](https://github.com/fidor/request_headers_logger/blob/master/lib/request_headers_logger.rb#L14)  is just the store items after filtering the whitelisted ones.

**Comment by MarcGrimme** [comment]
:+1:

**Pull request merged**